distributed/scheduler.py

```diff
@@ -8408,3 +8452,50 @@ def transition(
                 self.metadata[key] = ts.metadata
                 self.state[key] = finish
                 self.keys.discard(key)
+
+
+def coassignmnet_groups(
+    tasks: Sequence[TaskState], start: int = 0
```

Comment:
OK, so tasks is a list of taskstates sorted in increasing priority order.

Reply:
yes, I wanted to add a TODO to verify this but this is guaranteed in update_graph so for this prototype, it works